trieste/acquisition/multi_objective/pareto.py

```diff
@@ -73,6 +79,184 @@ def hypervolume_indicator(self, reference: TensorType) -> TensorType:
+        elif n_samples_difference > 0:
+            # We need to randomly add samples
+            samples_to_incr = np.random.choice(
+                np.arange(len(n_times_sampled)), size=n_samples_difference
+            )
+            for idx in samples_to_incr:
+                n_times_sampled[idx] += 1
+
+        # Create a list of the sample indices
+        sample_ids = list()
+        for idx, repeats in enumerate(n_times_sampled):
+            for _ in range(int(repeats)):
+                sample_ids.append(idx)
+        # Convert to array for indexing and return
+        sample_ids_array = np.array(sample_ids)
+
+        # Create batch with each of the selected samples
+        samples = np.array(self.front)[sample_ids_array]
+
+        return samples, sample_ids_array
+
+    def _choose_batch_no_repeats(
+        self, x_star: TensorType, sample_size: int
+    ) -> tuple[TensorType, TensorType]:
+
+        front_size = self.front.shape[0]
+
+        # Create id array to keep track of points
+        id_arr = np.expand_dims(np.arange(front_size), axis=1)
+
+        # Stitch id array, x_star and the front together
+        stitched_array = np.concatenate([id_arr, x_star, np.array(self.front)], axis=1)
+
+        # Sort array by x_star descending
+        sorted_array = stitched_array[stitched_array[:, 1].argsort()[::-1]]
+
+        samples = sorted_array[:sample_size, 2:]
+        sample_ids = sorted_array[:sample_size, 0].astype(int)
+
+        return samples, sample_ids
+
+    def _find_x_star(self, q: TensorType, p: TensorType) -> TensorType:
+
+        front_size = self.front.shape[0]
+
+        p_diag = np.expand_dims(np.diagonal(p), axis=1)
+
+        # Solve quadratic program for y*
+        P = cp.atoms.affine.wraps.psd_wrap(q)
+        G = np.eye(front_size)
+        h = np.zeros(front_size)
+        A = np.transpose(p_diag)
+        b = np.ones(1)
+
+        # Define and solve the CVXPY problem.
+        y = cp.Variable(front_size)
+        prob = cp.Problem(cp.Minimize((1 / 2) * cp.quad_form(y, P)), [G @ y >= h, A @ y == b])
+        prob.solve()
+
+        y_star = y.value
+
+        # Calculate x*
+        x_star = np.expand_dims(y_star, axis=1) / np.sum(y_star)
+
+        return x_star
+
+    def _calculate_p_matrix(
+        self, lower_bound: TensorType, reference_point: TensorType
+    ) -> TensorType:
+
+        front_size, front_dims = self.front.shape
+
+        p = np.zeros([front_size, front_size])
+
+        # Calculate denominator value for p matrix elements
+        denominator: float = 1
+        for i in range(front_dims):
+            if reference_point[i] - lower_bound[i] == 0:
+                raise ValueError(
+                    "Pareto set has identical upper and lower bounds"
+                    " in a dimension, this means you either have multiples"
+                    " of a single point, or only one point"
+                )
+            denominator *= reference_point[i] - lower_bound[i]
```

Comment:
Will crash later if they are equal - by construction they shouldn't, unless your Pareto front is a single point where this would happen. In that case your delta is zero and your bounds are equal. 

Single point Pareto front is not a corner case so we should have something that works.